packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1868,6 +1871,212 @@ function commitPassiveLifeCycles(finishedWork: Fiber): void {
   }
 }
 
+function commitDoubleInvokeEffectsInDEV(
+  fiber: Fiber,
+  hasPassiveEffects: boolean,
+) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      invokeLayoutEffectsUnmountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsUnmountInDEV(fiber);
+      }
+
+      invokeLayoutEffectsMountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsMountInDEV(fiber);
+      }
+    }
+  }
+}
+
+function invokeLayoutEffectsUnmountInDEV(firstChild) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      // unmount layout effects
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          // Should we add a separate subtree tag for this?
+          const primarySubtreeTag = fiber.subtreeTag & LayoutSubtreeTag;
+          if (primarySubtreeTag !== NoSubtreeTag) {
+            invokeLayoutEffectsUnmountInDEV(fiber.child);
+          }
+        }
+
+        const effectTag = fiber.effectTag;
+        const current = fiber.alternate;
+        // This is a mount
+        if (current === null) {
+          if (effectTag & Update) {
+            switch (fiber.tag) {
+              case FunctionComponent:
+              case ForwardRef:
+              case SimpleMemoComponent:
+              case Block: {
+                invokeGuardedCallback(
```

Comment:
Technically, after https://github.com/facebook/react/pull/19605/files this will be unnecessary since commitHookEffectListUnmount should be safe now.

It's an awkward inconsistency since the equivalent mount isn't safe.